### ISSUE TITLE
Instructions for setting up private key based authorization?

### ISSUE THREAD

**Issue by KarloX2** [PowerShell/Win32-OpenSSH#1318]
**"OpenSSH for Windows" version**
 7.6.0.1
**Server OperatingSystem**
Windows 10 Pro

Hi!

I'd really need a step-by-step guide that instructs me how I can setup/configure the OpenSSH server so that a SSH client can be authenticated using a private key file, without being asked to enter a password. I stumbled across threads that mention the Authentication Agent (what is it actually doing?) and command lines tools like ssh-add.exe (where does it add what to?) and ssh-keygen.exe etc. I played with it but I just don't see any results. I'm fishing in the dark.

Could someone provide the steps or is there already a guide for that?

Thanks so much!
Karlo

**Comment by manojampalam** [comment]
Did you see
https://github.com/PowerShell/Win32-OpenSSH/wiki/ssh.exe-examples

It should be straight forward to set it up. If you need to trouble shoot see:
https://github.com/PowerShell/Win32-OpenSSH/wiki/Troubleshooting-Steps

**Comment by KarloX2** [comment]
Hello @manojampalam 

Thanks for pointing me to this Wiki. Essentially, I think this is what I tried before, but it doesn't work as expected.

Here's what I did (client and server are the same machine at the moment):

- Started OpenSSH Authentication Agent
- delete all priviously added keys: `ssh-add -D`
- create a new private/public keys pair in `C:\users\myusername\.ssh` by giving this command: `ssh-keygen -t rsa -f id_rsa` (with an empty password when asked)
- Added the key by typing `ssh-add id_rsa` which gives the message `Identity added: id_rsa (id_rsa)`
- Copied `id_rsa.pub` to the sub directory (newly created) `C:\users\myusername\.ssh\authorized_keys`
- Restart both the OpenSSH Authentication Agent and the OpenSSH SSH Server services (just in case)
- Try to login with `ssh -i .\id_rsa myusername\localhost`
- Result: I can log in, but ssh still asks me for the password (blank isn't accepted, I must enter the user's password) wich is what I want to get rid of.


Then I tried to add the setting `AuthenticationMethods publickey` in sshd_config and restartet the services. Now when I try `ssh -i .\id_rsa myusername\localhost` it fails with the message `strobel@localhost: Permission denied (publickey).`


What can I try next?
sshd.exe is of version 7.6.0.1. Do I need to update (and how)?

Thanks
